test/integration/agent-integration.test.ts

```diff
@@ -130,6 +134,30 @@ describe('Integration test', () => {
       })
     })
 
+    it('properly captures protected resources', async () => {
+      const username = 'test'
+      const password = 'test'
+
+      app.get('/auth/redirected.png', (_, res) => {
+        res.redirect(301, '/auth/fairy-emojione.png')
+      })
+
+      app.use(
+        '/auth',
+        basicAuth({ users: { [username]: password }, challenge: true }),
+        express.static(testCaseDir),
+      )
+
+      await page.authenticate({ username, password })
+      await page.goto(`http://localhost:${PORT}/auth/protected-with-basic-auth.html`)
+
+      await snapshot(page, 'Protected assets', {
+        requestHeaders: {
+          Authorization: `Basic ${Buffer.from(`${username}:${password}`).toString('base64')}`,
```

Comment:
What do you think about adding another snapshot that _doesn't_ pass the auth headers?

Reply:
Added. I actually tested this TDD style by making sure it failed before it passed 👍